### PR TITLE
Fixing docs to remove references to links under terms/

### LINKED ISSUE #15530: Remove the terms folder
* Replace any links to the files in /terms with links to reference/glossary.md
* Remove the /terms directory

### PR DESCRIPTION
**Pull request by dharmit**
Fixes #15530 

Removed references to links under `terms/` directory and the directory itself.

### CODE REVIEW

File: docs/reference/builder.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ be UPPERCASE in order to distinguish them from arguments more easily.
 
 Docker runs the instructions in a `Dockerfile` in order. **The
 first instruction must be \`FROM\`** in order to specify the [*Base
-Image*](/terms/image/#base-image) from which you are building.
+Image*](/reference/glossary/#base-image) from which you are building.
```
Comment (thaJeztah) [inline docs/reference/builder.md]: Not your fault, but looking at the definition of "base image" in the glossary, I think that definition doesn't apply here. The  definition of "Parent Image" (in the old "terms" document) is a better match here. *However* most people speak of "base image" when talking about `FROM`.

@moxiegirl halp! any ideas how to solve this?

Comment (moxiegirl) [inline docs/reference/builder.md]: @thaJeztah @dharmit is fixing the issue as stated. To fix the change, requires adding a new definition and then linking to it.  I think that should be done in another PR.  The new glossary term would need to go through tech review.

**Comment by moxiegirl** [comment]
@dharmit nicely done for your first time contribution. I've checked your work -- looks good. Apologies for the delay in getting my response. LGTM

@thaJeztah if you are good with my response, LGTM.

**Comment by thaJeztah** [comment]
@moxiegirl yeah I'm good; thought we could possibly reuse some of the text we're removing in this PR, hence my slight "halp"

LGTM and thanks so much for contributing @dharmit. Congratulations with your first merge! :tada:

**Comment by dharmit** [comment]
@moxiegirl Thank you so much for the kind words. Can't wait to pickup another issue and contribute more!

@thaJeztah Thank you! I hope to convert this into first of many more to come! :D
